### PR TITLE
Add new-changelog to Makefile

### LINKED ISSUE #7865: Makefile target to help create changelog file.
**Describe the problem/challenge you have**

Makes it easier to correctly add new unreleased changelogs to correct dir.

**Describe the solution you'd like**

A placeholder file that will persists the changelogs/unreleased dir

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by kaovilai**
Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change
Make it easier to create new changelog file correctly.

Example usage:

```
~/git/velero makefile-changelog*
❯ make new-changelog 
"Add new-changelog to Makefile" added to ./changelogs/unreleased/8104-kaovilai

~/git/velero makefile-changelog*
❯ make new-changelog CHANGELOG_BODY="Changes you have made" 
"Changes you have made" added to ./changelogs/unreleased/8104-kaovilai
```

# Does your change fix a particular issue?

Fixes #7865

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -377,4 +377,20 @@ test-perf: local
...
-	go generate ./pkg/...
\ No newline at end of file
+	go generate ./pkg/...
+
+# requires an authenticated gh cli
+# gh: https://cli.github.com/
+# by default uses PR title as changelog body but can be overwritten like so
+# make new-changelog CHANGELOG_BODY="Changes you have made"
+new-changelog: GH_LOGIN ?= $(shell gh pr view --json author --jq .author.login 2> /dev/null)
+new-changelog: GH_PR_NUMBER ?= $(shell gh pr view --json number --jq .number 2> /dev/null)
```
Comment (reasonerjt) [inline Makefile:387]: I never used `gh` cmd, but is this always reliable?

Comment (kaovilai) [reply Makefile:387]: It's reliable. It's GitHub's own CLI.

Comment (kaovilai) [reply Makefile:387]: It gets the right PR number for me.

I could add more vars here for base branch for those who work on other downstream but for folks with their fork and tanzu this should work out of the box.

Comment (kaovilai) [reply Makefile:387]: The alternative is to curl against GitHub rest API

Comment (kaovilai) [reply Makefile:387]: It's based on the current branch you checked out. GitHub will check if the branch is used as a PR against upstream.

Comment (sseago) [reply Makefile:387]: @kaovilai we'd need to document somewhere that users need to install the github CLI, though.

Comment (kaovilai) [reply Makefile:387]: Line 382 count?

Comment (sseago) [reply Makefile:387]: Oh, yeah -- that's probably good enough. And this is an optional task that's not called anywhere else, so if it's not there, it won't matter as long as this task isn't executed explicitly.

**Review by sseago — APPROVED** [review]

**Comment by kaovilai** [comment]
FYI: I don't really need this any more.. especially if no one else find use of it.
I have made it a zsh function in my own dotfiles so it is not necessary to merge this. :)

**Comment by kaovilai** [comment]
https://github.com/kaovilai/dotfiles/blob/master/zsh/util.zsh#L21

**Comment by blackpiglet** [comment]
This PR is valuable to me, but I think we need to document more details on how to use it.

In my test, it seems need to create a PR first, then run the added make command.

```
gh pr create --title 'Title name' --body 'PR body' --label 'label-name-1'

make new-changelog
```

File: Makefile [hunk 2]
```diff
@@ -377,4 +377,20 @@ test-perf: local
 	$(MAKE) -e VERSION=$(VERSION) -C test/ run-perf
 
 go-generate:
-	go generate ./pkg/...
\ No newline at end of file
+	go generate ./pkg/...
+
+# requires an authenticated gh cli
+# gh: https://cli.github.com/
+# by default uses PR title as changelog body but can be overwritten like so
```
Comment (kaovilai) [inline Makefile:384]:
```suggestion
# First create a PR
# gh pr create --title 'Title name' --body 'PR body' --label 'label-name-1'
# by default uses PR title as changelog body but can be overwritten like so
```

**Comment by kaovilai** [comment]
@blackpiglet done.

**Review by blackpiglet — APPROVED** [review]

**Comment by blackpiglet** [comment]
Thanks for your effort.
The content looks good. 
Please squash the commits and resolve the DCO error.

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**